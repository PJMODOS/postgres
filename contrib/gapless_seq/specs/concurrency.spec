setup
{
	CREATE EXTENSION IF NOT EXISTS gapless_seq;
    DROP SEQUENCE IF EXISTS test_gapless;
	CREATE SEQUENCE test_gapless USING gapless;
}

teardown
{
    DROP SEQUENCE test_gapless;
}

session "s1"
step "s1_begin" { BEGIN; }
step "s1_nextval" { SELECT nextval('test_gapless'::regclass); }
step "s1_commit" { COMMIT; }

session "s2"
step "s2_begin" { BEGIN; }
step "s2_nextval" { SELECT nextval('test_gapless'::regclass); }
step "s2_commit" { COMMIT; }

session "s3"
step "s3_begin" { BEGIN; }
step "s3_nextval" { SELECT nextval('test_gapless'::regclass); }
step "s3_rollback" { ROLLBACK; }

permutation "s1_begin" "s1_nextval" "s2_begin" "s2_nextval" "s1_commit" "s2_commit"
permutation "s3_begin" "s3_nextval" "s2_begin" "s2_nextval" "s3_rollback" "s2_commit"
