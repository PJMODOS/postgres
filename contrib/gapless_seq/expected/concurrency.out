Parsed test spec with 3 sessions

starting permutation: s1_begin s1_nextval s2_begin s2_nextval s1_commit s2_commit
step s1_begin: BEGIN;
step s1_nextval: SELECT nextval('test_gapless'::regclass);
nextval        

1              
step s2_begin: BEGIN;
step s2_nextval: SELECT nextval('test_gapless'::regclass); <waiting ...>
step s1_commit: COMMIT;
step s2_nextval: <... completed>
nextval        

2              
step s2_commit: COMMIT;

starting permutation: s3_begin s3_nextval s2_begin s2_nextval s3_rollback s2_commit
step s3_begin: BEGIN;
step s3_nextval: SELECT nextval('test_gapless'::regclass);
nextval        

1              
step s2_begin: BEGIN;
step s2_nextval: SELECT nextval('test_gapless'::regclass); <waiting ...>
step s3_rollback: ROLLBACK;
step s2_nextval: <... completed>
nextval        

1              
step s2_commit: COMMIT;
